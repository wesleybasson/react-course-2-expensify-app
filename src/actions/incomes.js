import database from '../firebase/firebase';

//ADD INCOME
export const addIncome = (income) => ({
    type: 'ADD_INCOME',
    income
})

export const startAddIncome = (incomeData = {}) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid
        const {
            description = '',
            note = '',
            amount = 0,
            createdAt = 0
        } = incomeData
        const income = { description, note, amount, createdAt }
        return database.ref(`users/${uid}/incomes`).push(income)
            .then((ref) => {
                dispatch(addIncome({
                    id: ref.key,
                    ...income
                }))
            })
    }
}

//REMOVE INCOME
export const removeIncome = ({ id } = {}) => ({
    type: 'REMOVE_INCOME',
    id
})

export const startRemoveIncome = ({ id } = {}) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid
        return database.ref(`users/${uid}/incomes/${id}`).remove().then(() => {
            dispatch(removeIncome({ id }))
        })
    }
}

//EDIT INCOME
export const editIncome = (id, updates) => ({
    type: 'EDIT_INCOME',
    id,
    updates
})

export const startEditIncome = (id, updates) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid
        return database.ref(`users/${uid}/incomes/${id}`).update(updates).then(() => {
            dispatch(editIncome(id, updates))
        })
    }
}

//SET INCOMES
export const setIncomes = (incomes) => ({
    type: 'SET_INCOMES',
    incomes
})

export const startSetIncomes = () => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid
        return database.ref(`users/${uid}/incomes`)
        .once('value', (snapshot) => {
            const incomes = []
            snapshot.forEach((childSnapshot) => {
                incomes.push({
                    id: childSnapshot.key,
                    ...childSnapshot.val()
                })
            })
            dispatch(setIncomes(incomes))
        })
    }
}