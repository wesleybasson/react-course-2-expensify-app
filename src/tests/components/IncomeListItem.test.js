import React from 'react';
import { shallow } from 'enzyme';
import IncomeListItem from '../../components/IncomeListItem';
import incomes from '../fixtures/incomes';

test('should render IncomeListItem with fixture income data', () => {
    const wrapper = shallow(<IncomeListItem key={incomes[1].id} {...incomes[1]} />);
    expect(wrapper).toMatchSnapshot();
});