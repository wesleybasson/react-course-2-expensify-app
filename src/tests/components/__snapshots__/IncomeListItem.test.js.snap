// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render IncomeListItem with fixture income data 1`] = `
<Link
  className="list-item"
  replace={false}
  to="/editincome/2"
>
  <div>
    <h3
      className="list-item__title"
    >
      Rent Income
    </h3>
    <span
      className="list-item__subtitle"
    >
      December 28th, 1969
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    $2,345.00
  </h3>
</Link>
`;
